pwm/hostname.sls

```diff
@@ -0,0 +1,47 @@
+# This is currently intended for ec2 hosts that come with sth like ip-10-10-0-123 for a hostname.
+# To match the hostname with the entries created by the hostsfile state, also execute this state.
+
+{%- set fqdn = grains['nodename'] %}
```

Comment:
This file seems like it's working awfully hard to set the hostname. Should start with something designed to set the hostname, and then extend from there:

```
system:
  network.system:
    - hostname: {{ fqdn }}
    - apply_hostname: True
    - retain_settings: True
```